tests/v3_api/test_auth.py

```diff
@@ -233,6 +291,234 @@ def login(username, password, expected_status=201):
     return token
 
 
+def enable_openldap(username, token, expected_status=200):
+    headers = {'Authorization': 'Bearer ' + token}
+    r = requests.put(CATTLE_AUTH_PROVIDER_URL, json={
+      "ldapConfig": {
+        "accessMode": "unrestricted",
+        "connectionTimeout": CONNECTION_TIMEOUT,
+        "certificate": CA_CERTIFICATE,
+        "groupDNAttribute": "entryDN",
+        "groupMemberMappingAttribute": "member",
+        "groupMemberUserAttribute": "entryDN",
+        "groupNameAttribute": "cn",
+        "groupObjectClass": "groupOfNames",
+        "groupSearchAttribute": "cn",
+        "nestedGroupMembershipEnabled": False,
+        "port": PORT,
+        "servers": [
+          HOSTNAME_OR_IP_ADDRESS
+        ],
+        "serviceAccountDistinguishedName": SERVICE_ACCOUNT_NAME,
+        "tls": TLS,
+        "userDisabledBitMask": 0,
+        "userLoginAttribute": "uid",
+        "userMemberAttribute": "memberOf",
+        "userNameAttribute": "cn",
+        "userObjectClass": "inetOrgPerson",
+        "userSearchAttribute": "uid|sn|givenName",
+        "userSearchBase": USER_SEARCH_BASE,
+        "serviceAccountPassword": SERVICE_ACCOUNT_PASSWORD
+      },
+      "enabled": True,
+      "username": username,
+      "password": PASSWORD
+    }, verify=False, headers=headers)
+    assert r.status_code == expected_status
+    print "Enable openLdap request for " + username + " " + str(expected_status)
+    print r.json()
+
+
+def disable_openldap(username, token, expected_status=200):
+    headers = {'Authorization': 'Bearer ' + token}
+    r = requests.put(CATTLE_AUTH_PROVIDER_URL, json={
+      "ldapConfig": {
+        "accessMode": "unrestricted",
```

Comment:
Do you have to POST the entire config to diable an auth provider? Just posting the "enabled=False" flag in ldapConfig{} does not work?

Reply:
We have the "disable" action on the authconfig, we should be disabling using that API